### DISCUSSION TITLE
This seems like a bad idea.

### DISCUSSION THREAD

**Discussion by jhilgeman** [spatie/invade#3] [General]
Just my two cents - if someone is willing to download a separate utility class to save them from writing less than 10 lines of documented reflection code, they're probably the type of developer who shouldn't be messing with private properties in the first place. 

Anyone who is using reflection to interact with inaccessible scopes should be extremely aware that they're using it, since they're literally trying to bypass the entire point of something being private (or protected) to begin with. More often than not, that leads to unintended behavior and obscure support requests.

**Comment by freekmurze** [comment]
I respectfully disagree with this stance. I'm always assuming that other programmers know what they are doing when they use special stuff like this.

A paragraph from the Python guide that resonates well with me.
https://python-guide-chinese.readthedocs.io/zh_CN/latest/writing/style.html#we-are-all-consenting-adults

**Reply by yakatz** [reply]
A comment in the README that this might not be a good idea for production use still might be a good idea. This would be similar to the python documentation you referenced.
I can see how this could be useful for testing though.

**Comment by gilbertoalbino** [comment]
LoL! What's next in this PHP's OOP Book? 
Make all methods and properties public!
Why bother?

**Comment by gilbertoalbino** [comment]
Code Smell won't fit here, but Debug Smell? What about?

**Comment by avidianity** [comment]
This is actually quite useful in testing. I'm using it in one of my laravel projects right now.

**Reply by gilbertoalbino** [reply]
Hope you know how to use a Debugger instead of requiring packages into you projects!

**Reply by jhilgeman** [reply]
Can you elaborate on how you're using it in a way that wouldn't be handled more appropriately by implementing getters/setters? 

If it's only being used for testing, then what kind of testing are you doing that is useful but doesn't reflect how the code will run in production?
